conans/util/templates.py

```diff
@@ -1,6 +1,24 @@
-# coding=utf-8
+"""
+Gather all the utilities related to Jinja2 templates.
+"""
 
-from jinja2 import Template
+import os
+
+from jinja2 import Environment, FileSystemLoader, ChoiceLoader, select_autoescape, Template
+
+from conans.assets.templates import dict_loader
+
+
+def get_template(template_name, templates_folder):
+    # Loader order: first current working dir, then 'templates' folder, fallback to dictionary
+    loader = ChoiceLoader([
+        FileSystemLoader(os.getcwd()),
```

Comment:
I would love to move the `env` object into the `app.cache`  one, so it is created only once and it can take advantage of internal cache. Not really important right now, the Conan client is running only one command each time it is loaded... but from the `conan_api` POV it would be important.

Reply:
If we agree on moving this to the `app.cache` I'll make a PR with that refactor.

Reply:
Maybe I would wait for this functionality of different templates. I would focus first on the replacement of the current table, and will wait for this, as I understand this will be used also for other templates, like the graph one.